web/pdf_link_service.js

```diff
@@ -493,6 +493,48 @@ class PDFLinkService {
     });
   }
 
+  /**
+   * @param {Object} action
+   */
+  async executeSetOCGState(action) {
+    const pdfDocument = this.pdfDocument;
+    const optionalContentConfig = await this.pdfViewer
+      .optionalContentConfigPromise;
+
+    if (pdfDocument !== this.pdfDocument) {
+      return; // The document was closed while the optional content resolved.
+    }
+    let operator;
+
+    for (const elem of action.state) {
+      switch (elem) {
```

Comment:
I don't really see why this `switch` and the `operator` variable are necessary. Simply switching on `elem` instead of `operator` below should do the same as far as I can tell: only ON/OFF/Toggle are handled anyway and any other value simply does nothing.

Reply:
*Sorry, I forgot to include a spec-link above.*

As you can see in the example in https://web.archive.org/web/20220309040754if_/https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf#G11.1990554 there can be *multiple* layer-Refs after any "operator", which thus (as far as I can tell) require us to track of the last seen operator.

Reply:
Ah, right, that explains it indeed. Thanks!